fastly/resource_fastly_tls_subscription.go

```diff
@@ -294,10 +294,35 @@ func resourceFastlyTLSSubscriptionRead(_ context.Context, d *schema.ResourceData
 	if err != nil {
 		return diag.FromErr(err)
 	}
+
+	// The configuration ID returned by the API is the initial TLS configuration when first
+	// activated. Instead, this value should represent the current TLS configuration id which
+	// can be found via the following API call.
+	// https://api.fastly.com/tls/domains?filter[tls_certificates.id]={cert-id}&include=tls_activations
 	err = d.Set("configuration_id", subscription.Configuration.ID)
 	if err != nil {
 		return diag.FromErr(err)
 	}
+	var tlsDomains []*gofastly.TLSDomain
+	tlsDomains, err = conn.ListTLSDomains(&gofastly.ListTLSDomainsInput{
+		FilterTLSCertificateID: certificateID,
+		Include:                "tls_activations",
+	})
+	for _, tlsDomain := range tlsDomains {
+		// Activations may be empty (omitempty)
+		if tlsDomain.Activations == nil {
+			break
+		}
+		len := len(tlsDomain.Activations)
+		if len > 0 {
+			err = d.Set("configuration_id", tlsDomain.Activations[len-1].Configuration.ID)
```

Comment:
I prefer not to shadow a builtin function (i.e. `len()`)...

```suggestion
		activations := len(tlsDomain.Activations)
		if activations > 0 {
			err = d.Set("configuration_id", tlsDomain.Activations[activations-1].Configuration.ID)
```

Reply:
@virgofx UPDATE!

I've spoken with the TLS team and they recommended sorting the `tlsDomain.Activations` by `CreatedAt` as this will ensure the `[activations-1]` will return the relevant action object.